### PR TITLE
[UG]Enhancements aimed for future iterations

### PR DESCRIPTION
**Pull request by Ma-Xizi**

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -315,6 +317,40 @@ the acceptable range). Therefore, edit the data file only if you are confident t
...
+To support easy organization and access, we’re introducing a categorization system that allows you to sort students by their enrolled subjects and grade levels. 
+This enhancement will help you quickly filter students and manage their academic needs more efficiently.
+
+#### 3. **Exam Date Management**
+We understand the importance of staying prepared for exams, so we’re adding a feature to track upcoming exam dates for each student. 
+With this addition, guardians and you will receive timely reminders or notifications, ensuring students are ready and supported in their preparation.
+
+#### 4. **Fee and Payment Tracking**
+Keeping track of fees is essential in the business aspect, so we plan to implement a detailed payment tracking system. 
+This feature will log due dates, total payment amounts, and any outstanding fees for each student, while also linking this information to the responsible guardian.
```
Comment (Ma-Xizi) [inline docs/UserGuide.md:338]: These are from past user stories, and not specific to any PE-Dru bug reports.

File: docs/UserGuide.md [hunk 2]
```diff
@@ -113,6 +113,7 @@ Remarks:
     * end with a `domain` label at least 2 characters long
     * have each `domain` label start and end with alphanumeric characters
     * have each `domain` label consist of alphanumeric characters, separated only by hyphens, if any.
+* Duplicated emails is strictly not allowed.
```
Comment (Chronoxy) [inline docs/UserGuide.md:116]: There are some merge conflicts here. I have made changes in PR #308

**Review by Chronoxy — COMMENTED** [review]
Overall LGTM! but need to fix merge conflicts and let's keep this PR until all future enhancements are updated 👍

File: docs/UserGuide.md [hunk 3]
```diff
@@ -315,6 +317,47 @@ the acceptable range). Therefore, edit the data file only if you are confident t
...
+Recognizing that most tuition lessons occur weekly at the same time, we plan to introduce a feature that allows you to set lessons as either one-time sessions or recurring events. 
+You will be able to stipulate the frequency of recurrence, whether weekly, bi-weekly, or custom intervals, to fit their specific scheduling needs.
+
+#### 6. **Overlapping Lesson Time Handling**
+We recognize the need for clarity around overlapping lesson scheduling. Options for managing overlaps will include:
+
+* Allowing overlapping sessions for group lessons or pair lessons.
+* Enabling you to customize settings for overlap permissions based on the type of lesson.
+
+#### 7. **Enhanced Name Field Support**
```
Comment (ethantyh692) [inline docs/UserGuide.md:350]: I think for this we need to include the handling of white space?

See #270

Comment (ethantyh692) [reply docs/UserGuide.md:350]: I have added that issue under this PR

File: docs/UserGuide.md [hunk 4]
```diff
@@ -315,6 +317,47 @@ the acceptable range). Therefore, edit the data file only if you are confident t
...
+To accommodate contacts with names that contain special characters, such as commas, we plan to update the application to allow names with these characters. 
+In the future, the system will support a wider variety of name formats, making it more inclusive and user-friendly.
+
+#### 8. **Past Lesson Management**
+To enhance lesson organization, we plan to introduce features that clearly distinguish past lessons. This will include options to automatically hide completed lessons or display them in a different color for easy identification.
+You will also be able to toggle the visibility of past lessons, allowing them to focus on upcoming schedules while still being able to reference completed sessions if needed.
+
+#### 9. **Support for Decimal Places in Fees**
```
Comment (ethantyh692) [inline docs/UserGuide.md:358]: Perhaps here we can include the example outlined in #281 to make it very explicit.

As that issue is also covered here I will add that to the PR

**Pull request merged**